packages/mds-ingest-service/service/validators.ts

```diff
@@ -89,8 +90,42 @@ const telemetrySchema = Joi.object<TelemetryDomainModel>()
   })
   .unknown(false)
 
-export const { validate: validateTelemetryDomainModel, isValid: isValidTelemetryDomainModel } =
-  schemaValidator<DeviceDomainModel>(telemetrySchema)
+const nullableInteger = { type: 'integer', nullable: true, default: null }
+
+const nullableFloat = { type: 'number', format: 'float', nullable: true, default: null }
+
+export const { validate: validateTelemetryDomainModel, $schema: TelemetrySchema } =
+  SchemaValidator<TelemetryDomainModel>(
+    {
+      $id: 'Telemetry',
+      type: 'object',
+      properties: {
+        device_id: uuidSchema,
+        provider_id: uuidSchema,
+        timestamp: { type: 'integer' },
+        gps: {
+          type: 'object',
+          properties: {
+            lat: { type: 'number', format: 'float' },
+            lng: { type: 'number', format: 'float' },
+            // ⬇⬇⬇ NULLABLE/OPTIONAL PROPERTIES ⬇⬇⬇
+            altitude: nullableFloat,
+            heading: nullableFloat,
+            speed: nullableFloat,
+            accuracy: nullableFloat,
+            hdop: nullableFloat,
+            satellites: nullableInteger
+          },
+          required: ['lat', 'lng']
+        },
+        // ⬇⬇⬇ NULLABLE/OPTIONAL PROPERTIES ⬇⬇⬇
+        charge: { type: 'number', format: 'float', minimum: 0, maximum: 1.0, nullable: true, default: null },
```

Comment:
Could this  just add `minimum/maximum` to `{ …nullableFloat }`?

Reply:
It could indeed!